p1-staking-pool-dyn/src/lib.rs

```diff
@@ -121,25 +116,18 @@ impl Contract {
         assert!(amount >= MIN_STAKE, "{}", ERR01_MIN_STAKE);
         let aid = env::predecessor_account_id();
         self.total_stake += amount;
-        match self.vaults.get(&aid) {
-            Some(mut vault) => {
-                self._stake(amount, &mut vault);
-                self.vaults.insert(&aid, &vault);
-                return vault.staked.into();
-            }
-            None => {
-                self.vaults.insert(
-                    &aid,
-                    &Vault {
-                        // warning: previous can be set in the future
-                        previous: cmp::max(current_round(), self.farming_start),
-                        staked: amount,
-                        rewards: 0,
-                    },
-                );
-                return amount.into();
-            }
-        };
+        let mut vault = self.get_vault_or_default(&aid);
+        if vault.is_empty() {
+            // new account no need to ping & _stake
+            vault.staked += amount;
+            // warning: previous can be set in the future
+            vault.previous = cmp::max(current_round(), self.farming_start);
+        } else {
+            self._stake(amount, &mut vault);
+        }
+        // save vault
+        self.save_vault(&aid, &vault);
+        return vault.staked.into();
```

Comment:
why doing this change? I think the original version was more clear: the default is not used and we overwrite it anyway.